templates/telemetry/normandy-login-study/normandy-login-study.4.schema.json

```diff
@@ -0,0 +1,87 @@
+{
+  "$schema": "http://json-schema.org/draft-04/schema#",
+  "type": "object",
+  "title": "normandy-login-study",
+  "properties": {
+    "password_manager_enabled": {
```

Comment:
Pings in the `telemetry` namespace and sent via the desktop telemetry library generally have some common top-level elements like environment, clientId, etc. injected into the template. See for example https://github.com/mozilla-services/mozilla-pipeline-schemas/blob/master/templates/telemetry/testpilot/testpilot.4.schema.json#L6-L10

If this ping doesn't need those things, though, by all means don't include them.

The rest of the body of the ping is expected to be nested under a `payload` field and use camelCase field names.

That said, the structure you already have here looks pretty appropriate for a "structured ingestion" ping that lives in a namespace other than `telemetry`. In that case, the client-side implementation would not use the desktop telemetry library and would instead send pings explicitly to the pipeline; this is how new tab page activity-stream pings are implemented for desktop Firefox, for example.

Reply:
This ping does not have any of those additional fields that are typical of most telemetry pings (they are not included on purpose). The json provided in `normandy-login-study.4.sample.pass.json` shows the complete contents of an actual test ping. 

@ericawright can you fill in which library you are using to actually send the ping? 

If she is not using the desktop telemetry libraries, would she still need to modify the ping to be nested under `payload`?  

I can change the namespace once I know what would be appropriate to use.

Reply:
@jklukas actually I think we are using this library to send the ping from the addon https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/collection/webextension-api.html

Reply:
```
  const payload = {
    password_manager_enabled,
    [...]
    history_average_days_per_month,
  };

  // addClientId defaults to false, being explicit.
  browser.telemetry.submitPing("normandy-login-study", payload, {addClientId: false});
```
This uses the webextension api for telemetry
https://firefox-source-docs.mozilla.org/toolkit/components/telemetry/collection/webextension-api.html

Reply:
Happy to nest things under a "payload" key, or change the keys to camelCase if required.

Reply:
What we are sending is already nested under `payload` if you look at the raw ping sent to the server.

Reply:
Here is a really raw ping JSON for my profile:
```json
{
  "type": "normandy-login-study",
  "id": "0714c230-13ec-e04b-85cf-8fd3fc381b91",
  "creationDate": "2020-06-19T03:32:47.518Z",
  "version": 4,
  "application": {
    "architecture": "x86-64",
    "buildId": "20200617215206",
    "name": "Firefox",
    "version": "79.0a1",
    "displayVersion": "79.0a1",
    "vendor": "Mozilla",
    "platformVersion": "79.0a1",
    "xpcomAbi": "x86_64-gcc3",
    "channel": "nightly"
  },
  "payload": {
    "password_manager_enabled": true,
    "browser_privatebrowsing_autostart": false,
    "network_cookie_lifetime_policy": 0,
    "privacy_sanitize_sanitize_on_shutdown": false,
    "privacy_clear_on_shutdown_cookies": false,
    "network_cookie_cookie_behavior": 3,
    "browser_startup_page": 3,
    "session_days_old": 1,
    "profile_days_old": 3178,
    "logins_accounts": true,
    "logins_accounts_uses_per_month": 0.1,
    "google_accounts_cookie_present": true,
    "google_accounts_cookie_days_old": 878,
    "has_allow_cookie_exceptions": false,
    "has_block_cookie_exceptions": false,
    "cookies_oldest_days_old": 1243,
    "activeAddons": [
      {
        "id": "tabstats@glandium.org",
        "name": "Tab Stats"
      },
      {
        "id": "jid0-ePBKtqqSeez8ThB1NInNoY7B0xc@jetpack",
        "name": "Better MXR"
      },
      {
        "id": "jid0-NgMDcEu2B88AbzZ6ulHodW9sJzA@jetpack",
        "name": "BugzillaJS"
      },
      {
        "id": "{bb682c45-3136-4213-bf29-5f5833080bf4}",
        "name": "Context Plus"
      },
      {
        "id": "adb@mozilla.org",
        "name": "Firefox DevTools ADB Extension"
      },
      {
        "id": "superstop@gavinsharp.com",
        "name": "SuperStop"
      },
      {
        "id": "nl-NL@dictionaries.addons.mozilla.org",
        "name": "Woordenboek Nederlands"
      },
      {
        "id": "{09420bd6-3c35-444d-b85c-74d51fe3df4f}",
        "name": "Discard Tab"
      },
      {
        "id": "jid1-KR7YJG7ASAuPHQ@jetpack",
        "name": "TabSubmit Basic"
      },
      {
        "id": "en-CA@dictionaries.addons.mozilla.org",
        "name": "Canadian English Dictionary"
      },
      {
        "id": "doh-rollout@mozilla.org",
        "name": "DoH Roll-Out"
      },
      {
        "id": "tabcenter-reborn@ariasuni",
        "name": "Tab Center Reborn"
      },
      {
        "id": "aboutsync@mhammond.github.com",
        "name": "About Sync"
      },
      {
        "id": "{91aa5abe-9de4-4347-b7b5-322c38dd9271}",
        "name": "Clippings"
      },
      {
        "id": "redirector@einaregilsson.com",
        "name": "Redirector"
      },
      {
        "id": "@testpilot-containers",
        "name": "Firefox Multi-Account Containers"
      },
      {
        "id": "@contain-facebook",
        "name": "Facebook Container"
      },
      {
        "id": "https-everywhere-eff@eff.org",
        "name": "HTTPS Everywhere"
      },
      {
        "id": "uBlock0@raymondhill.net",
        "name": "uBlock Origin"
      },
      {
        "id": "fr-dicollecte@dictionaries.addons.mozilla.org",
        "name": "Dictionnaire français"
      },
      {
        "id": "alltabshelper@alltabshelper.org",
        "name": "All Tabs Helper"
      },
      {
        "id": "{b9acf540-acba-11e1-8ccb-001fd0e08bd4}",
        "name": "Easy Youtube Video Downloader Express"
      },
      {
        "id": "default-theme@mozilla.org",
        "name": "Default"
      },
      {
        "id": "amazondotcom@search.mozilla.org",
        "name": "Amazon.com"
      },
      {
        "id": "ebay@search.mozilla.org",
        "name": "eBay"
      },
      {
        "id": "google@search.mozilla.org",
        "name": "Google"
      },
      {
        "id": "wikipedia@search.mozilla.org",
        "name": "Wikipedia (en)"
      },
      {
        "id": "ddg@search.mozilla.org",
        "name": "DuckDuckGo"
      },
      {
        "id": "bing@search.mozilla.org",
        "name": "Bing"
      },
      {
        "id": "secure-proxy@mozilla.com",
        "name": "Firefox Private Network"
      },
      {
        "id": "{bf855ead-d7c3-4c7b-9f88-9a7e75c0efdf}",
        "name": "Zoom Scheduler"
      },
      {
        "id": "screenshots@mozilla.org",
        "name": "Firefox Screenshots"
      },
      {
        "id": "webcompat-reporter@mozilla.org",
        "name": "WebCompat Reporter"
      },
      {
        "id": "formautofill@mozilla.org",
        "name": "Form Autofill"
      },
      {
        "id": "webcompat@mozilla.org",
        "name": "Web Compat"
      },
      {
        "id": "login-study@mozilla.org",
        "name": "Google Accounts Login Check"
      },
      {
        "id": "Shockwave FlashShockwave Flash 32.0 r0",
        "name": "Shockwave Flash"
      },
      {
        "id": "gmp-gmpopenh264",
        "name": "OpenH264 Video Codec provided by Cisco Systems, Inc."
      },
      {
        "id": "gmp-widevinecdm",
        "name": "Widevine Content Decryption Module provided by Google Inc."
      }
    ],
    "default_search_engine_is_google": false,
    "default_private_search_engine_is_google": false,
    "accounts_days_visited_per_month": 5.61,
    "has_browser_search_with_ads": true,
    "has_browser_search_ad_clicks": true,
    "history_oldest_days_old": 453,
    "history_average_days_per_month": 25.72
  }
}
```

Reply:
thanks @mnoorenberghe I will update the PR

Reply:
Glad to hear there's already an implementation that we can make sure to match. To be clear, the camelCase vs. snake_case thing is fine. We can write the schema to match what's being sent here.